### DISCUSSION TITLE
Strange behaviour in Wireshark

### DISCUSSION THREAD

**Discussion by aggedor2** [thirtythreeforty/neolink#220] [Q&A]
I'm trying to sniff communication between a rebranded Reolink Argus 2 using Wireshark.

My problem is that absolutely nothing comes in Wireshark up when I view the camera from the smartphone app.

I tried filtering on the IP address (that the app tells me is registered to the camera) while viewing the camera with the app. There are no transactions with that ip address appearing in the capture.

The only time it appears is if I ping that IP address from my laptop.

I also tried filtering on its Ethernet Mac address. Still nothing is shown in Wireshark (apart from the pings I ran).

I checked on my router. In the router device list, it shows the Mac address for the camera and that it's active, but it doesn't show the DHCP address it assigned.

Is it possible that they are using a new protocol that isn't IP? Would that be shown in Wireshark?

The only thing that came up at all in Wireshark (apart from pings) were broadcasts from the Android app to 255.255.255.255. These get reported by the baichuan dissector.

Any pointers on what's going on?

**Comment by QuantumEntangledAndy** [comment]
Let's start with the basics. Where is your wireshark running on?

1. The smartphone
2. Another computer on the network
3. The router

If your answer is 2. Then that is your issue. You cannot spy on information that does not come to you.

Smart phone sends packets to the router, router sends packets to camera. A laptop/computer sees nothing. the only thing that the computer would see is the broadcasts which the router sends to everyone.

**Reply by twistedddx** [reply]
*the switch sends the broadcast to everyone.

Edit: Although I now realize that might have been simplified language as it is common in small networks to call a device that is a router, wireless access point and switch - simply a "router".

**Reply by QuantumEntangledAndy** [reply]
Yes technically the switch I apologies if I overly glossed over the details. I have a tendency to do that at times. Thanks for the assist, and making sure I don't mix things up too much :)

**Comment by aggedor2** [comment]
Thanks for clarifying, that's what's happening then. How do you get all the
activity on the local network then using a sniffer? Do I have to hack the
router and run it there?

**Reply by aggedor2** [reply]
So is it just a matter of running Wireshark in promiscuous mode, that allows me to see all network packets? I'm on M1 Mac + Monterey, so I guess I'll need to work out whether that's possible or broken. Is anyone here running Wireshark in promiscuous mode on Mac?

**Reply by thirtythreeforty** [reply]
Maybe, maybe not. If your router is smart enough to support "mirroring" traffic then you can get it to send you everything that happens on a port.

Even if you're in promiscuous mode, there are a variety of scenarios where sniffing wifi is not going to work super well. The bulletproof way is to use an app on the same PC. Another way may be to use an app (maybe [this one](https://play.google.com/store/apps/details?id=com.emanuelef.remote_capture)? Haven't tried it) that pretends to be a VPN and logs all the traffic to a pcap file you can load in Wireshark.

Mind telling the brand name of the camera? I was not aware Reolink white-labels their stuff.

**Reply by twistedddx** [reply]
You can buy an old skool hub.
If the device has a light of collisions, you are looking at the right device.

**Reply by QuantumEntangledAndy** [reply]
You can set your mac to share it's internet connection. Then get the phone and camera to connect through that (providing it's a wifi camera). Then the mac will be acting as a "router".

**Reply by aggedor2** [reply]
> Maybe, maybe not. If your router is smart enough to support "mirroring" traffic then you can get it to send you everything that happens on a port.
> 
> Even if you're in promiscuous mode, there are a variety of scenarios where sniffing wifi is not going to work super well. The bulletproof way is to use an app on the same PC. Another way may be to use an app (maybe [this one](https://play.google.com/store/apps/details?id=com.emanuelef.remote_capture)? Haven't tried it) that pretends to be a VPN and logs all the traffic to a pcap file you can load in Wireshark.
> 
> Mind telling the brand name of the camera? I was not aware Reolink white-labels their stuff.

Thanks. Discovered their Mac client so that simplifies things, I'll just run client and Wireshark on the same machine. The cameras are from Uniden - App Cam Solo+, App Cam Solo 4G, etc

**Reply by twistedddx** [reply]
Unidens support page has a link to an old style client.
Uniden 11.2.2.31 should be equal to Reolink 7.2.2.31. The last Reolink client I saw for the old style was 7.2.2.33

Uniden client is using encryption 01 which is perfect for checking things with.

https://uniden.com.au/product/guardian-app-cam-spotlight-plus/

**Comment by Lanwalker** [comment]
I have been monitoring your conversations about your wireshark issues... Try using an inexpensive WIFI extender and have the wireless cameras connect thru it and either use a wired port on the extender or have your capture station connect to it by WIFI also. 

Network switches are an odd fellow when trying to do capturing and most are ' intelligent ' enough to not waste the bandwidth to send another stations traffic out to the capture device and makes it difficult to see everything happening. 

An inexpensive network hub rarely has the multi band protocols to use wire and wireless to be able to separate the traffic but adding the extender as an extra hop and then connecting the extender to the hub may give you more flexibility to see what you are looking for, especially by also having the capture system and the camera connecting thru it.

'Sniffers ' are usually fairly expensive devices as are the routers that have the capability to ' mirror ' the traffic out of another switch port which would be an ideal environment for your debug setups at home... using an extender may increase your chances of making your home network a little easier to test and capture information, especially if you also connect it thru an inexpensive hub instead of a switched hub system.

I use 3 different Netgear extenders on my home network to make sure the cameras can connect back to my internal home router to increase my confidence the cameras can reach that distance but it is a little tricky to make multiple extenders from Netgear work on my home network... There wasn't a way to initially change the extender networks name ( they seem to like using your original router's network name and adding what band to the end in their setup ) so I took the first extender off, installed the 2nd using their setup and then renamed it afterwards and did the same for the next extender to make it work. Also, it only had a single wired port if you wanted to use wire to extend the distance as a bridge or connect a wired extension device out that far.

This little device I just found on Amazon sounds interesting if I was going to do it again ( I really like using wire if possible for the additional security ) since it has 2 wired ports...

WiFi Extender, Up to 1200Mbps WiFi Range Repeater 2.4Ghz & 5Ghz Dual Band Covers Up to 1500 Sq.ft and 25 Devices Wireless Signal Booster with Ethernet Port (Black)
Brand: TRANGU 
( Save 25%  Lowest price in 30 days
Was:    
$39.99
$39.99 Details
Price:    
$29.99
$29.99 FREE One-Day & FREE Returns 
You Save:    
$10.00
$10.00 (25% ) )
https://www.amazon.com/Extender-1200Mbps-Repeater-Wireless-Ethernet/dp/B094ZL3JVK/ref=sr_1_1_sspa?crid=1FMSP5ADT9MLX&keywords=wifi+extender+with+ethernet+port&qid=1636285408&sprefix=Wifi+extender%2Caps%2C199&sr=8-1-spons&psc=1&spLa=ZW5jcnlwdGVkUXVhbGlmaWVyPUEyMVlMWkowTkM0VUxRJmVuY3J5cHRlZElkPUEwMDM0MDY2MzhXN0pINVhRSTFIUSZlbmNyeXB0ZWRBZElkPUEwMjUzNzg5Mk9GWTBTUTQ0STlYSCZ3aWRnZXROYW1lPXNwX2F0ZiZhY3Rpb249Y2xpY2tSZWRpcmVjdCZkb05vdExvZ0NsaWNrPXRydWU=
